devices/cpu/registers.c

```diff
@@ -19,20 +19,21 @@
 #include "registers.h"
 #include "../../debugger/io.h"
 
+#define OPCODE_MASK 0xFFC0u
+#define OPCODE_CALL_INSTRUCTION 0x1280u
+
 //##########+++ MSP430 Register initialization +++##########
-void initialize_msp_registers(Emulator *emu)
+void initialize_msp_registers(Emulator* const emu)
 {
   Cpu *cpu = emu->cpu;
   Debugger *debugger = emu->debugger;
 
-  /* Initialize PC to boot loader code on cold boot (COLD)*/
-  //cpu->pc = 0x0C00;
-
   /* Initialize Program Counter to *0xFFFE at boot or reset (WARM)*/
   cpu->pc = 0xC000;
 
-  /* Stack pointer typically begins at the top of RAM after reset */
-  cpu->sp = 0x400;
+  /* Stack pointer - set to the end of the address space, should be set by
+     the program */
+  cpu->sp = 0xFFFF;
```

Comment:
End of the address space? There should be address of RESET ISR or something  like this. Could you explain?

Reply:
It is explained -> this should be initialized by the program. 0xFFFF is less random than 0x400, which may or may not be true, depending on various things,
There is also a motive for a change -> maximum stack size tracing is easier and more coherent, when we can just use min function instead of a pair of min/max, to catch the actual stack initialization.